src/app/src/components/annotations/annotator.tsx

```diff
@@ -652,22 +662,43 @@ export default class Annotator extends Component<
     this.filterAnnotationVisibility();
   };
 
-  private updateImage = () => {
+  private updateCacheList = async () => {
+    if (this.props.loadedModel) {
+      await APIGetCacheList(this.props.loadedModel.hash)
+        .then(res => {
+          this.setState({ cacheList: res.data });
+        })
+        .catch(error => {
+          let message = `Failed to load cached predictions. ${error}`;
+          if (error.response) {
+            message = `${error.response.data.error}: ${error.response.data.message}`;
+          }
+          CreateGenericToast(message, Intent.DANGER, 3000);
+        });
+    }
+  };
+
+  private updateImage = async () => {
+    await this.updateCacheList();
+
     /* Get All Existing Registered Folder and Image Assets */
-    APIGetAsset().then(res => {
+    await APIGetAsset().then(res => {
       /* Generate New Asset List Based on Updated Data */
       const newImageAssets = res.data.map((encodedUri: string) => {
         const decodedUri = decodeURIComponent(encodedUri);
         let seperator = "/";
```

Comment:
same as below, try to use const and ternary

Reply:
done